src/NLog/Targets/DatabaseTarget.cs

```diff
@@ -712,28 +726,83 @@ protected string BuildConnectionString(LogEventInfo logEvent)
             sb.Append("Server=");
             sb.Append(RenderLogEvent(DBHost, logEvent));
             sb.Append(";");
-            if (DBUserName == null)
+            var dbUserName = RenderLogEvent(DBUserName, logEvent);
+            if (string.IsNullOrEmpty(dbUserName))
             {
                 sb.Append("Trusted_Connection=SSPI;");
             }
             else
             {
                 sb.Append("User id=");
-                sb.Append(RenderLogEvent(DBUserName, logEvent));
+                sb.Append(dbUserName);
                 sb.Append(";Password=");
-                sb.Append(RenderLogEvent(DBPassword, logEvent));
+                var password = GetRenderedAndEscapedPassword(logEvent);
+                sb.Append(password);
                 sb.Append(";");
             }
-
-            if (DBDatabase != null)
+            var dbDatabase = RenderLogEvent(DBDatabase, logEvent);
+            if (!string.IsNullOrEmpty(dbDatabase))
             {
                 sb.Append("Database=");
-                sb.Append(RenderLogEvent(DBDatabase, logEvent));
+                sb.Append(dbDatabase);
             }
 
             return sb.ToString();
         }
 
+        private string GetRenderedAndEscapedPassword(LogEventInfo logEvent)
+        {
+            if (_fixedDbPassword != null)
+            {
+                return _fixedDbPassword;
```

Comment:
Like this optimization :)

Reply:
Thanks, I was also working on better approach (adapter instead of 2 fields), but that will be another PR

Reply:
Thanks, I was also working on better approach (adapter instead of 2 fields), but that will be another PR